lib/SILOptimizer/Mandatory/DataflowDiagnostics.cpp

```diff
@@ -46,16 +46,17 @@ static void diagnoseMissingReturn(const UnreachableInst *UI,
     llvm_unreachable("unhandled case in MissingReturn");
   }
 
-  // No action required if the function returns 'Void' or that the
-  // function is marked 'noreturn'.
-  if (ResTy->isVoid() || F->isNoReturnFunction())
+  // No action required if the function returns 'Void'.
+  if (ResTy->isVoid())
```

Comment:
Hm... I'm seeing what effect removing this would have, but you're right it seems this code path is unexercised.